interfaces/builtin/maliit_input_method.go

```diff
@@ -0,0 +1,155 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var maliitInputMethodPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as a maliit server.
+
+# DBus accesses
+#include <abstractions/dbus-session>
+
+/usr/share/glib-2.0/schemas/ r,
+
+# maliit uses peer-to-peer dbus over a unix socket after address negotiation
+# each application has its own one-to-one communication channel with the maliit
+# server, over which all further communication happens.
+unix (bind, listen, accept) type=stream addr="@/tmp/maliit-server/dbus-*",
+`)
+
+var maliitInputMethodConnectedSlotAppArmor = []byte(`
+# Provides the maliit address service which assigns an individual unix socket
+# to each application
+dbus (send, receive)
+    bus=session
+    interface="org.maliit.Server.Address"
+    path=/org/maliit/server/address
+    peer=(label=###PLUG_SECURITY_TAGS###),
```

Comment:
I'm not clear on why this works then (but it does), because the order of communication is that the application first connects to the "org.maliit.Server.Address" interface on the session bus to then be assigned a unix socket. I don't get any apparmor denials about this, and communication happens without any problems, so I'm a bit confused as to what's happening here.